common/keyboard.go

```diff
@@ -219,14 +249,7 @@ func (k *Keyboard) keyDefinitionFromKey(key keyboardlayout.KeyInput) keyboardlay
 	if srcKeyDef.Text != "" {
 		keyDef.Text = srcKeyDef.Text
 	}
-
-	strKey := string(key)
-	if strings.ToUpper(strKey) == strKey && srcKeyDef.ShiftKey != "" {
-		keyDef.Key = srcKeyDef.ShiftKey
-		keyDef.Text = srcKeyDef.ShiftKey
-	}
-
-	if shift != 0 && srcKeyDef.ShiftKey != "" && strings.Contains(string(key), "Key") {
+	if _, ok := keyAToKeyZ[string(key)]; (ok || foundInShift) && shift != 0 && srcKeyDef.ShiftKey != "" {
```

Comment:
Can you move this logic (`(ok || foundInShift) && shift != 0 && srcKeyDef.ShiftKey != "")`) into a closure? It's harder to see what it does.